internal/common/pointers.go

```diff
@@ -3,3 +3,13 @@ package common
 func ToPtr[T any](x T) *T {
 	return &x
 }
+
+// DerefDefault returns the dereferenced value of the given pointer or the
+// default value for the type if unset.
+func DerefDefault[T any](p *T) T {
```

Comment:
Maybe `DerefOrDefault` (i.e. *Or* in the middle?) and maybe a small unit test. 

Fwiw, I wish we wonder if we could consider a more holistic approach to this problem, I opened https://github.com/osbuild/images/commit/fc3af96416cb9a3ee47938d7494843ceb74f8c72 a while ago for discussion which tries to tackle a similar issue. But its fine of course, there are tradeoffs and this is nice and simple.

Reply:
Yeah, that is much nicer.  Not sure why it stalled back then tbh.